grpclog/grpclog.go

```diff
@@ -50,6 +50,15 @@ func Infoln(args ...interface{}) {
 	logger.Infoln(args...)
 }
 
+// InfoDepth logs to the INFO log at the specified depth.
+func InfoDepth(depth int, args ...interface{}) {
```

Comment:
Let's move these to `internal/grpclog` so they are not visible to users.

To avoid the circular dependency (`grpclog` depends on `internal/grpclog` and vice-versa) and also avoid the type assertion where this is used:

`package internal/grpclog`:
- Copy `DepthLogger` and `V2Logger` interfaces here (unfortunate but this is internal).  Type aliases _would_ be better, but those will make the godoc of `grpclog` bad.
- Add globals: `var DepthLogger DepthLogger` and `var Logger V2Logger`.
- Add `func InfoDepth`/etc, using `DepthLogger` if non-nil, or calling `Logger.Info`/etc otherwise.
  - Our packages call these functions instead of via the `grpc/grpclog` package.

`package grpclog`:  imports `internal/grpclog`
- Remove global `logger`.
- Set  `internal/grpclog.Logger` and `internal/grpclog.DepthLogger` together, clearing the `DepthLogger` if the `V2Logger` does not implement it.
- Use `internal/grpclog.Logger` instead of the global `logger`.